### PR TITLE
Create Class Card Component

### LINKED ISSUE #14: Allow users to add and drop classes they are taking
## Description
Create Class component -- a button with the name of the course and a delete button.

## Learning objectives
- React
- API Wrapper
- CSS/Flexbox
- Frontend

## Todo
- Create a component called `Class.jsx` in `\client\src\components\ that displays a class name in the format "CS125"` and has a delete button on it
- Add functions in API wrapper that calls backend endpoint to edit a user (PUT) (this is not add/delete because we are only adding and removing from a field in the User database (`classes` array), not adding/deleting an actual User)

### PR DESCRIPTION
**Pull request by eeshabarua**
## Status:

🚀 

## Description

I wrote a function in apiWrapper to call a put request on the classes field in User. 

Fixes #14 

## Todos

I want to stylize the card css to create a right-side delete icon and clean up the feature. I also want to build out the functionality of the put request in the frontend.

## Screenshots

image
image

### CODE REVIEW

File: client/src/css/Classes.scss [hunk 1]
```diff
@@ -0,0 +1 @@
+@import '../styles/colors.scss';
```
Comment (dyang5200) [inline client/src/css/Classes.scss]: Remove this file since it's not being used.

File: client/src/components/ClassCard.jsx [hunk 2]
```diff
@@ -0,0 +1,22 @@
+import React from 'react';
+import { Card, Button } from 'semantic-ui-react';
+
+import { putUserClass } from '../utils/apiWrapper';
+
+// import '../css/Class.scss';
```
Comment (dyang5200) [inline client/src/components/ClassCard.jsx]: Remove comment

File: client/src/utils/apiWrapper.js [hunk 3]
```diff
@@ -39,3 +39,18 @@ export const addSampleResponse = (body) => {
       error,
     }));
 };
+
+//calling backend endpoint to edit user class in profile
+export const putUserClass = (body) => {
```
Comment (dyang5200) [inline client/src/utils/apiWrapper.js]: give this a more descriptive name. putUserClass isn't intuitive on the frontend. change to `editUserClasses` or something similar

File: client/src/components/ClassCard.jsx [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+
+function ClassCard(props) {
+  const { text } = props;
+  return (
+    <Card style={{ height: '85px' }}>
```
Comment (dyang5200) [inline client/src/components/ClassCard.jsx]: For best practices, you should never have inline styling. Extract styles to a separate scss file

File: client/src/components/ClassCard.jsx [hunk 5]
```diff
@@ -0,0 +1,22 @@
+import React from 'react';
+import { Card, Button } from 'semantic-ui-react';
+
+import { putUserClass } from '../utils/apiWrapper';
+
+// import '../css/Class.scss';
+
+function ClassCard(props) {
+  const { text } = props;
```
Comment (dyang5200) [inline client/src/components/ClassCard.jsx]: This variable name is not descriptive enough. `Text` is ambiguous; give it a more descriptive name that describes what the text is

File: client/src/components/ClassCard.jsx [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+  return (
+    <Card style={{ height: '85px' }}>
+      <Card.Content>
+        <Card.Description>{text}</Card.Description>
+        <Button basic color="red" onclick={putUserClass()}>
+          Delete
+        </Button>
```
Comment (dyang5200) [inline client/src/components/ClassCard.jsx]: Change `Delete` to an `x` for aesthetic purposes. Also, the delete button should be on the same line as the button text. Another thing, the syntax for onclick functions are like this: `onClick`. Make sure to test your code and look up documentation to make sure all of it is working as you intend it.

File: client/src/pages/Profile.jsx [hunk 7]
```diff
@@ -1,9 +1,19 @@
...
 function Profile() {
   return (
     <>
       <h1>Profile Page</h1>
+      <div className="class-grid-container">
+        <ClassCard text={'CS125'} />
+        <ClassCard text={'CS128'} />
+        <ClassCard text={'CS173'} />
+        <ClassCard text={'CS222'} />
+        <ClassCard text={'CS225'} />
```
Comment (dyang5200) [inline client/src/pages/Profile.jsx]: This is repetitive code. Please use a .map() function to modularize your code like we talked about in lessons.

**Review by dyang5200 — CHANGES_REQUESTED** [review]
Good start Eesha! Looks like a few things still need to be fixed before this is ready. Let me know if you have any questions!

File: client/src/utils/apiWrapper.js [hunk 8]
```diff
@@ -39,3 +39,18 @@ export const addSampleResponse = (body) => {
       error,
     }));
 };
+
+//calling backend endpoint to edit user class in profile
+export const editUserClasses = (body) => {
```
Comment (ashwin1104) [inline client/src/utils/apiWrapper.js:44]: sweet

File: client/src/css/ClassCard.scss [hunk 9]
```diff
@@ -0,0 +1,4 @@
+.card-content-container {
+    display: flex;
+    justify-content: space-between;
+}
```
Comment (ashwin1104) [inline client/src/css/ClassCard.scss]: Add a new line here. It's formatting standards to have a new line at the end of each file

File: client/src/components/ClassCard.jsx [hunk 10]
```diff
@@ -0,0 +1,20 @@
...
+import '../css/ClassCard.scss';
+
+
+function ClassCard(props) {
+  const { classCardText } = props;
+  return (
+    <Card>
+      <Card.Content className="card-content-container">
+        <Card.Description>{ classCardText }</Card.Description>
+        <Button basic color="red" onClick={ editUserClasses() }>X</Button>
```
Comment (ashwin1104) [inline client/src/components/ClassCard.jsx]: Change `onClick={ editUserClasses() }` to `onClick={editUserClasses}` (i.e. remove the parentheses). Let's discuss the reasoning in person

File: client/src/components/ClassCard.jsx [hunk 11]
```diff
@@ -0,0 +1,20 @@
...
+import { Card, Button } from 'semantic-ui-react';
+
+import { editUserClasses } from '../utils/apiWrapper';
+import '../css/ClassCard.scss';
+
+
+function ClassCard(props) {
+  const { classCardText } = props;
+  return (
+    <Card>
```
Comment (ashwin1104) [inline client/src/components/ClassCard.jsx]: Make the styling changes I suggested for a cleaner look:
- shrink the X button and move it closer to the right-hand-side
- enlarge the classCardText
- center the classCardText in the card

ClassCardBasicSketch

**Review by ashwin1104 — CHANGES_REQUESTED** [review]
Nice work Eesha, very clean code! Left some important changes for you to address. Also change the title of your PR to the commit message conventions we mentioned (starting with an imperative verb, briefly describe purpose of PR). Example: Create ClassCard component.

**Review by dyang5200 — APPROVED** [review]
Great :)

**Review by ashwin1104 — APPROVED** [review]

**Pull request merged**